### PR TITLE
Add tests for changes due to bidi/whitespace

### LINKED ISSUE #888: Add tests for changes due to bidi/whitespace #884
Update the tests to handle bidi controls and spacing changes (should be none of the latter)

### PR DESCRIPTION
**Pull request by catamorphism**
Closes #888

### CODE REVIEW

File: test/tests/bidi.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    },
+    {
+        "description": "complex-message   = o *(declaration o) complex-body o",
+        "src": ".local $x = {1} \u200F {{ {$x}}}",
+        "exp": " 1"
+    },
+    {
+        "description": "complex-message   = o *(declaration o) complex-body o",
+        "src": ".local $x = {1} {{ {$x}}} \u2066",
+        "exp": "1"
```
Comment (aphillips) [inline test/tests/bidi.json:26]: Missing a space before the output, since the pattern is `{{ {$x}}}`

**Review by aphillips — CHANGES_REQUESTED** [review]
One regular-issue bug, I think? Otherwise looks good.

Might be good to have some of the examples from the design doc using the strict flavor as examples. I can make a separate PR for that, though.

File: test/tests/bidi.json (hunk 1 above)
Comment (catamorphism) [reply test/tests/bidi.json]: Oops, there were actually a few more of these, because I just found a bug in my ICU4C test runner. Fixed in [f82e12a](https://github.com/unicode-org/message-format-wg/pull/902/commits/f82e12a8a9ca7bfeafcb041c0ed02a94de90c7a9)

**Review by aphillips — APPROVED** [review]

File: test/tests/bidi.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    {
+        "description": "match-statement s variant *(o variant)",
+        "src": ".local $x = {1 :number}.match $x \u061c1 {{one}}* {{other}}",
+        "exp": "one"
+    },
+    {
+        "description": "match-statement s variant *(o variant)",
+        "src": ".local $x = {1 :number}.match $x\u061c1 {{one}}* {{other}}",
+        "exp": "one"
+    },
```
Comment (eemeli) [inline test/tests/bidi.json:68]: This is wrong; the space after `.match $x` is required: https://github.com/unicode-org/message-format-wg/blob/e9fc1367cbf496897736e68b0a29fd3ace229759/spec/message.abnf#L17

**Review by eemeli — CHANGES_REQUESTED** [review]

File: test/tests/bidi.json (hunk 2 above)
Comment (catamorphism) [reply test/tests/bidi.json]: Fixed in [fb13a36](https://github.com/unicode-org/message-format-wg/pull/902/commits/fb13a36e930009c5d14e5833b3a54a06f3d86abb)

**Review by eemeli — APPROVED** [review]
